controllers/precache.go

```diff
@@ -298,18 +346,43 @@ func (r *ClusterGroupUpgradeReconciler) extractPrecachingSpecFromPolicies(
 			case utils.PolicyTypeClusterVersion:
 				image := object["spec"].(map[string]interface {
 				})["desiredUpdate"].(map[string]interface{})["image"]
-				if len(spec.PlatformImage) > 0 && spec.PlatformImage != image {
-					msg := fmt.Sprintf("Platform image must be set once, but %s and %s were given",
-						spec.PlatformImage, image)
-					meta.SetStatusCondition(&clusterGroupUpgrade.Status.Conditions, metav1.Condition{
-						Type:    utils.PrecacheSpecValidCondition,
-						Status:  metav1.ConditionFalse,
-						Reason:  "PlatformImageConflict",
-						Message: msg})
-					return *new(ranv1alpha1.PrecachingSpec), nil
+				if image != nil {
+					if len(spec.PlatformImage) > 0 && spec.PlatformImage != image {
+						msg := fmt.Sprintf("Platform image must be set once, but %s and %s were given",
+							spec.PlatformImage, image)
+						meta.SetStatusCondition(&clusterGroupUpgrade.Status.Conditions, metav1.Condition{
+							Type:    utils.PrecacheSpecValidCondition,
+							Status:  metav1.ConditionFalse,
+							Reason:  "PlatformImageConflict",
+							Message: msg})
+						return *new(ranv1alpha1.PrecachingSpec), nil
+					}
+					spec.PlatformImage = fmt.Sprintf("%s", image)
+				} else {
+					upstream := object["spec"].(map[string]interface {
+					})["upstream"].(string)
+					channel := object["spec"].(map[string]interface {
+					})["channel"].(string)
+					version := object["spec"].(map[string]interface {
+					})["desiredUpdate"].(map[string]interface{})["version"].(string)
+					var image string
+					if upstream != "" && channel != "" && version != "" {
+						image, err = r.getImageForVersionFromUpdateGraph(upstream, channel, version)
+					}
+
+					if err != nil || image == "" {
+						msg := "Platform image could not be inferred, either upstream/channel/version are missing or upstream endpoint is unreachable"
+						meta.SetStatusCondition(&clusterGroupUpgrade.Status.Conditions, metav1.Condition{
+							Type:    utils.PrecacheSpecValidCondition,
+							Status:  metav1.ConditionFalse,
+							Reason:  "PlatformImageInvalid",
+							Message: msg})
+						return *new(ranv1alpha1.PrecachingSpec), nil
+					}
+
+					spec.PlatformImage = image
```

Comment:
Could it be better if we just pass the actual error as the condition message?
```suggestion
					image, err := r.getImageForVersionFromUpdateGraph(upstream, channel, version)
					if err != nil {
						meta.SetStatusCondition(&clusterGroupUpgrade.Status.Conditions, metav1.Condition{
							Type:    utils.PrecacheSpecValidCondition,
							Status:  metav1.ConditionFalse,
							Reason:  "PlatformImageInvalid",
							Message: err.Error()})
						return *new(ranv1alpha1.PrecachingSpec), nil
					}
					spec.PlatformImage = image
```